relay-filter/src/generic_filters.rs

```diff
@@ -0,0 +1,106 @@
+//! Implements generic filtering based on the [`RuleCondition`] DSL.
+//!
+//! Multiple generic filters can be defined and they are going to be checked in FIFO order. The
+//! first one that matches, will result in the event being discarded with a [`FilterStatKey`]
+//! identifying the matching filter.
+
+use crate::{FilterStatKey, GenericFiltersConfig};
+use relay_event_schema::protocol::Event;
+use relay_protocol::RuleCondition;
+
+/// Version of the generic filters schema.
+pub const VERSION: u16 = 1;
+
+/// Checks events by patterns in their error messages.
+pub fn matches(event: &Event, condition: Option<&RuleCondition>) -> bool {
+    // TODO: the condition DSL needs to be extended to support more complex semantics, such as
+    //  collections operations.
+    condition.map_or(false, |condition| condition.matches(event))
+}
+
+/// Filters events by patterns in their error messages.
+pub fn should_filter(event: &Event, config: &GenericFiltersConfig) -> Result<(), FilterStatKey> {
+    for (filter_name, filter_config) in config.filters.0.iter() {
+        if !filter_config.is_empty() && matches(event, filter_config.condition.as_ref()) {
+            return Err(FilterStatKey::GenericFilter(filter_name.clone()));
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::generic_filters::should_filter;
+    use crate::{FilterStatKey, GenericFilterConfig, GenericFiltersConfig, OrderedFilters};
+    use relay_event_schema::protocol::{Event, LenientString};
+    use relay_protocol::Annotated;
+    use relay_protocol::RuleCondition;
+
+    fn mock_filters() -> Vec<(String, GenericFilterConfig)> {
+        vec![
+            (
+                "firstReleases".to_string(),
+                GenericFilterConfig {
+                    is_enabled: true,
+                    condition: Some(RuleCondition::eq("event.release", "1.0")),
+                },
+            ),
+            (
+                "helloTransactions".to_string(),
+                GenericFilterConfig {
+                    is_enabled: true,
+                    condition: Some(RuleCondition::eq("event.transaction", "/hello")),
+                },
+            ),
+        ]
+    }
+
+    #[test]
+    fn test_should_filter_event() {
```

Comment:
This test tries to validate multiple cases, could we separate them into different tests? See [relay coding guidelines](https://www.notion.so/sentry/Relay-Code-and-Architecture-4fac481cf27741aba54d39594f1cae58?pvs=4#fd14645580f848d9969d124a59cc3f7d).